homeassistant/components/backblaze/__init__.py

```diff
@@ -0,0 +1,106 @@
+"""The Backblaze integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import cast
+
+from b2sdk.v2 import B2Api, Bucket, InMemoryAccountInfo, exception
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryError, ConfigEntryNotReady
+
+from .const import (
+    CONF_APPLICATION_KEY,
+    CONF_BUCKET,
+    CONF_KEY_ID,
+    CONF_PREFIX,
+    DATA_BACKUP_AGENT_LISTENERS,
+    DOMAIN,
+)
+
+type BackblazeConfigEntry = ConfigEntry[BackblazeConfig]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: BackblazeConfigEntry) -> bool:
+    """Set up Backblaze from a config entry."""
+
+    info = InMemoryAccountInfo()
+    b2_api = B2Api(info)
+
+    data = cast(dict, entry.data)
```

Comment:
I don't think we need to cast this

Reply:
That looks strange because you are not indexing the config entry, you are indexing it's data, so that error suspects me that you do `entry[SOMETHING]`